jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -187,76 +187,88 @@ public ImageAndRegistryClient call()
-    // handling it normally
+    // special handling if we happen upon a manifest list, extract manifests handle them normally
     if (manifestTemplate instanceof V22ManifestListTemplate) {
-      manifestAndDigest =
-          obtainPlatformSpecificImageManifest(
-              registryClient, (V22ManifestListTemplate) manifestTemplate);
-      manifestTemplate = manifestAndDigest.getManifest();
+      Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
+      for (Platform platform : platforms) {
+        manifestAndDigestList.add(
+            obtainPlatformSpecificImageManifest(
+                registryClient, (V22ManifestListTemplate) manifestTemplate, platform));
+      }
+    } else {
+      manifestAndDigestList.add(manifestAndDigest);
     }
 
-    switch (manifestTemplate.getSchemaVersion()) {
-      case 1:
-        V21ManifestTemplate v21ManifestTemplate = (V21ManifestTemplate) manifestTemplate;
-        buildContext
-            .getBaseImageLayersCache()
-            .writeMetadata(
-                buildContext.getBaseImageConfiguration().getImage(), v21ManifestTemplate);
-        return JsonToImageTranslator.toImage(v21ManifestTemplate);
-
-      case 2:
-        eventHandlers.dispatch(
-            LogEvent.lifecycle("Using base image with digest: " + manifestAndDigest.getDigest()));
-        BuildableManifestTemplate buildableManifestTemplate =
-            (BuildableManifestTemplate) manifestTemplate;
-        if (buildableManifestTemplate.getContainerConfiguration() == null
-            || buildableManifestTemplate.getContainerConfiguration().getDigest() == null) {
-          throw new UnknownManifestFormatException(
-              "Invalid container configuration in Docker V2.2/OCI manifest: \n"
-                  + JsonTemplateMapper.toUtf8String(buildableManifestTemplate));
-        }
-
-        DescriptorDigest containerConfigurationDigest =
-            buildableManifestTemplate.getContainerConfiguration().getDigest();
-
-        try (ThrottledProgressEventDispatcherWrapper progressEventDispatcherWrapper =
-            new ThrottledProgressEventDispatcherWrapper(
-                progressEventDispatcher.newChildProducer(),
-                "pull container configuration " + containerConfigurationDigest)) {
-          String containerConfigurationString =
-              Blobs.writeToString(
-                  registryClient.pullBlob(
-                      containerConfigurationDigest,
-                      progressEventDispatcherWrapper::setProgressTarget,
-                      progressEventDispatcherWrapper::dispatchProgress));
-
-          ContainerConfigurationTemplate containerConfigurationTemplate =
-              JsonTemplateMapper.readJson(
-                  containerConfigurationString, ContainerConfigurationTemplate.class);
+    for (ManifestAndDigest<?> manifestAndDigestInstance : manifestAndDigestList) {
+      manifestTemplate = manifestAndDigestInstance.getManifest();
```

Comment:
As we discussed yesterday, I think now it makes sense to factor out the code inside the loop as a separate method.

Reply:
Refactoring Done